dss/api/bundles/__init__.py

```diff
@@ -351,11 +352,22 @@ def _idempotent_save(blobstore: BlobStore, bucket: str, key: str, data: dict) ->
         return False, existing_data == data
     else:
         # write manifest to persistent store
-        blobstore.upload_file_handle(
-            bucket,
-            key,
-            io.BytesIO(json.dumps(data).encode("utf-8")),
-        )
+        _d = json.dumps(data).encode("utf-8")
+        size = len(_d)
+        part_size = 16 * 1024 * 1024
+        if isinstance(blobstore, S3BlobStore) and size > part_size:
+            with io.BytesIO(_d) as fh:
+                multipart_parallel_upload(
+                    blobstore.s3_client,
+                    bucket,
+                    key,
+                    fh,
+                    size=size,
+                    part_size=part_size,
+                    parallelization_factor=20
```

Comment:
Should this default to a multiple of the system cpu count?  Or deleted and the function default used?

Reply:
It's probably reasonable to parameterize this in terms of Lambda memory.

I think this can be handled in a future PR.

Reply:
Sounds good, should a TODO be added?

Reply:
todo added